Source/SwiftLintFramework/Rules/CustomRules.swift

```diff
@@ -50,11 +50,25 @@ public struct CustomRules: Rule, ConfigurationProviderRule {
     public init() {}
 
     public func validateFile(file: File) -> [StyleViolation] {
-        if configuration.customRuleConfigurations.isEmpty {
+        var configurations = configuration.customRuleConfigurations
+
+        if configurations.isEmpty {
             return []
         }
 
-        return configuration.customRuleConfigurations.flatMap {
+        if file.path != nil {
+            configurations = configurations.filter {
+                let included = $0.included.pattern
+                if included.isEmpty { return true }
+
+                let pathMatch = file.path?.rangeOfString(included,
```

Comment:
`String.rangeOfString(_:options:range:locale:)` creates `NSRegularExpression` on every call.
We can reuse `$0.included` instead of that.